feed/script.js

```diff
@@ -0,0 +1,205 @@
+const container = document.querySelector('.container');
+const activityFeedContainer = document.getElementById(ACITIVITY_FEED_CONTAINER);
+const activityList = document.querySelector('.activity-list');
+const tabsList = document.querySelector('.tabs');
+const lastElementContainer = document.querySelector(LAST_ELEMENT_CONTAINER);
+
+let query = {};
+let newLink = '';
+let activityFeedPage = 0;
+let nextLink = '';
+let isDataLoading = false;
+let category = CATEGORY.ALL;
+
+const tabsData = [
+  { name: 'All', 'data-type': CATEGORY.ALL, class: 'active' },
+  { name: 'Task Requests', 'data-type': CATEGORY.TASK_REQUESTS },
+  { name: 'Extension Requests', 'data-type': CATEGORY.EXTENSION_REQUESTS },
+  { name: 'Task', 'data-type': CATEGORY.TASK },
+  { name: 'OOO Requests', 'data-type': CATEGORY.OOO },
+];
+
+tabsData.forEach((tab) => {
+  const listItem = document.createElement('li');
+  listItem.textContent = tab.name;
+  listItem.dataset.type = tab['data-type'];
+  if (tab.class) {
+    listItem.classList.add(tab.class);
+  }
+  tabsList.appendChild(listItem);
+});
+
+const tabs = document.querySelectorAll('.tabs li');
+
+tabs.forEach((tab) => {
+  tab.addEventListener('click', async function () {
+    tabs.forEach((t) => t.classList.remove('active'));
+    this.classList.add('active');
+    category = this.dataset.type;
+    changeFilter();
+    populateActivityFeed({ category });
+  });
+});
+
+const changeFilter = () => {
+  nextLink = '';
+  activityFeedContainer.innerHTML = '';
+};
+
+async function renderFeed() {
+  changeFilter();
+  populateActivityFeed({ category });
+  addIntersectionObserver();
+}
```

Comment:
- why are we making this function async? can we make individual functions async as needed?
- suggestion:
    - move the render function to the bottom of the script as it would make understanding the flow easier.

Reply:
Noted.